lib/h3.rb

```diff
@@ -142,6 +142,21 @@ def self.h3_to_geo_boundary(h3_index)
   end
 
   def self.hex_ranges(h3_set, k)
+    h3_range_indexes = hex_ranges_ungrouped(h3_set, k)
+    out = Hash.new { |h, k| h[k] = [] }
+    h3_range_indexes.each_slice(max_kring_size(k)).each do |indexes|
+      h3_index = indexes.first
+
+      out[h3_index] = 0.upto(k).map do |j|
```

Comment:
Let me propose a refactor here. First, let's define:

```ruby
def max_hex_ring_size(k)
  k.zero? ? 1 : 6*k
end
```

which comes from [this C comment](https://github.com/uber/h3/blob/36a2cebd1cd292cfcfe171626df666d1885c0f18/src/h3lib/lib/algos.c#L545). And I have verified (`mrs` stands for Max Ring Size):

![img_3134](https://user-images.githubusercontent.com/3387/50241092-e280a380-03c6-11e9-9b8f-4c188a21857d.jpg)

And then (untested):

```ruby
out[h3_index] = 1.upto(k).map do |j|
  start  = j == 0 ? 0 : max_kring_size(j-1)
  length = max_hex_ring_size(j)
  indexes.slice(start, length)
end
```

I believe `max_hex_range` in that loop is more intention revealing.

Reply:
Ok, done! I'll move `max_hex_ring_size` to a private method after this is merged and I rebase the `refactor` branch.